chapters/05-raster-vector.qmd

```diff
@@ -1,7 +1,797 @@
+fig
+```
+
+
+The following expression extracts elevation values from `srtm.tif` according to `zion_points`, using `Rasters.extract`.
+
+
+::: {.callout-danger}
+The API here is not great, can we do better?  It currently returns a vector of named tuples, which is not very convenient.
+
+One thought is to use a Tables.jl materializer to convert the result if possible.  I understand the desire to return the geometry values.  But there must be a better way than this.
+:::
+
+```{julia}
+result1 = Rasters.extract(src_srtm, zion_points; geometry = false) .|> first
+```
+
+The first argument is the raster from which to extract values, and the second is the vector object (or collection of objects) according to which to extract the values. 
+
+
+
+::: {.callout-note}
+**Rasters.jl** does not yet support interpolation, so the values extracted are the values of the nearest cell center.  
+This corresponds to `interpolate='nearest'` in the Python `rasterstats` package.
+:::
+
+
+Either way, the resulting object is a vector of raster values, corresponding to `zion_points`.
+For example, here are the elevations of the first five points.
+
+```{julia}
+result1[1:5]
+```
+
+To get a `GeoDataFrame` with the original points geometries (and other attributes, if any), as well as the extracted raster values, we can assign the extraction result into a new column.
+
+```{julia}
+zion_points[!, "elev1"] = result1
+zion_points
+```
+
+You can read from a single band by selecting the band in the Raster.  TODO finish this text
+
+### Extraction to lines {#sec-extraction-to-lines}
+
+
+Raster extraction is also applicable with line selectors.
+The typical line extraction algorithm is to extract one value for each raster cell touched by a line.
+However, this particular approach is not recommended to obtain values along the transects, as it is hard to get the correct distance between each pair of extracted raster values.
+
+For line extraction, a better approach is to split the line into many points (at equal distances along the line) and then extract the values for these points using the "extraction to points" technique (@sec-extraction-to-points).
+To demonstrate this, the code below creates (see @sec-vector-data for recap) `zion_transect`, a straight line going from northwest to southeast of the Zion National Park.
+
+```{julia}
+coords = [[-113.2, 37.45], [-112.9, 37.2]]
+zion_transect = GI.LineString(coords)
+```
+
+The utility of extracting heights from a linear selector is illustrated by imagining that you are planning a hike.
+The method demonstrated below provides an 'elevation profile' of the route (the line does not need to be straight), useful for estimating how long it will take due to long climbs.
+
+First, we need to create a layer consisting of points along our line (`zion_transect`), at specified intervals (e.g., `250`).
+To do that, we need to transform the line into a projected CRS (so that we work with true distances, in $m$), such as UTM.
```

Comment:
```suggestion
To do that, we need to transform the line into a projected CRS (so that we work with true distances, in $m$), such as [Universal Transverse Mercator](https://en.wikipedia.org/wiki/Universal_Transverse_Mercator_coordinate_system).